panel/src/components/Forms/Input/NumberInput.vue

```diff
@@ -69,20 +72,43 @@ export default {
     }
   },
   methods: {
-    focus() {
-      this.$refs.input.focus();
-    },
-    onInvalid() {
-      this.$emit("invalid", this.$v.$invalid, this.$v);
+    decimals() {
+      if (Math.floor(this.step) === this.step) {
+        return 0;
+      }
+
+      return this.step.toString().split(".")[1].length || 0;
     },
-    onInput(value) {
+    format(value) {
       // don't convert empty values or the beginning of
       // a negative number to a Number
       if (value !== null && value !== "" && value !== "-" && value !== "-0") {
         value = Number(value);
       }
 
-      this.$emit("input", value);
+      const decimals = this.decimals();
+
+      if (decimals) {
+        value = parseFloat(value).toFixed(decimals);
```

Comment:
same here in if and else

Reply:
And in fact we can save the else statement and just return the normal one last.

Reply:
I didn't realize that the format method can be a lot simpler when we only use it on change.